### ISSUE TITLE
Issues with C++ back end and async/jester

### ISSUE THREAD

**Issue by onionhammer** [nim-lang/Nim#3299]
I'm having issues running this minimal example on Linux (debian ARMHF), with latest devel branch. Tried both clang and gcc. Nim crashes during compilation with SIGSEGV.

```nim
import jester
import asyncdispatch, asyncnet

echo "Hello, world!"
```

With Jester, I get

```
Hint: cgi [Processing]
lib/pure/asyncdispatch.nim(1423, 7) Hint: Processing statusContent as an async proc. [User]
SIGSEGV: Illegal storage access. (Attempt to read from nil?)
```

Commenting out `import jester`, I get

```
/home/usr/projects/test/nimcache/stdlib_rawsockets.cpp:370:8: error: assigning to 'NCSTRING' (aka 'char *') from incompatible type 'const char *'
                LOC6 = gai_strerror(gairesult);
                     ^ ~~~~~~~~~~~~~~~~~~~~~~~
/home/usr/projects/test/nimcache/stdlib_rawsockets.cpp:514:8: error: assigning to 'NCSTRING' (aka 'char *') from incompatible type 'const char *'
                LOC5 = hstrerror(h_errno);
                     ^ ~~~~~~~~~~~~~~~~~~
/home/usr/projects/test/nimcache/stdlib_rawsockets.cpp:667:8: error: assigning to 'NCSTRING' (aka 'char *') from incompatible type 'const char *'
                LOC9 = inet_ntop(AF_INET6, ((void*) (addr6)), result->data, ((NI32) ((result ? result->len : 0))));
                     ^ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
3 errors generated.

```

Executing

```
nim cpp test.nim
```

Yields these errors

```
nim c test.nim
```

does not.

**Comment by dom96** [comment]
@Araq ping

**Comment by onionhammer** [comment]
Update: this happens on OSX as well.

**Comment by tmm1** [comment]
This should have been fixed in 2f7802bea3f27eae92b828b0cb3d49b451f10839

**Comment by onionhammer** [comment]
@tmm1 this is a compiler error, not a library error.
